fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -3640,6 +3718,7 @@ struct ShardedRocksDBKeyValueStore : IKeyValueStore {
 
 		try {
 			wait(self->writeThread->stop());
+			wait(self->compactionThread->stop());
```

Comment:
Will this block the shutdown too long? Do we need to wait for compaction to complete? On the other hand, if we don't wait, is there any way to prevent `nullptr` errors in the compaction thread?

Reply:
In AWS, compacting a 300MB shard takes around 3s. Since we don't need the return result from compaction, using a non-flow thread could be an option.